### PR TITLE
Replace non-ascii UTF-8 characters by Zs or spaces

### PR DESCRIPTION
**Pull request by mjambon**
As a preprocessing step before parsing, this attempts to replace characters outside the ascii range by ascii characters. If the unicode character "whitespace", it gets replaced by spaces, otherwise it gets replaced by Zs. The length of the replacement string should be equal to the original UTF-8 sequence being replaced in most cases (but it's not guaranteed due to Unicode having multiple ways of encoding the same character. See https://www.wikiwand.com/en/Unicode_equivalence#/Normalization).

### CODE REVIEW

**Comment by mjambon** [comment]
I didn't write proper unit tests for this. Also, the project doesn't build due to the need for libraries `uucp` and uutf`.

**Comment by mjambon** [comment]
This is for fixing https://github.com/returntocorp/semgrep/issues/997

File: h_program-lang/Parse_info.ml [hunk 1]
```diff
@@ -534,8 +534,9 @@ let complete_token_location_large filename table x =
  *    skip them easily with one Common.exclude
  *)
 let tokenize_all_and_adjust_pos file tokenizer visitor_tok is_eof =
- Common.with_open_infile file (fun chan -> 
-  let lexbuf = Lexing.from_channel chan in
+ Common.with_open_infile file (fun chan ->
+  let string = Unicode.UTF8.asciify chan in
```
Comment (aryx) [inline h_program-lang/Parse_info.ml:538]: maybe we can do this only if the file contains a unicode char in the first place?

Comment (aryx) [reply h_program-lang/Parse_info.ml:538]: I was just worried of slowing down a bit things only because of a few files. but maybe it is negligible.

**Pull request merged**